pkg/kubectl/apis/manifest/v1alpha1/types.go

```diff
@@ -0,0 +1,138 @@
+	// hosting the resource files specified in Base and Overlays.
+	// This is useful in the fork/rebase workflow.
+	Source string `json:"source,omitempty" yaml:"source,omitempty"`
+
+	// Version of the package.
+	PackageVersion string `json:"packageVersion,omitempty" yaml:"packageVersion,omitempty"`
+}
+
+// Manifest has all the information to expand of generate the k8s api resources.
+// It can be used by kubectl or some other tooling.
+// A manifest could be either a Base or an Overlay.
+type Manifest struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	// TODO: figure out if we need field ManifestVersion.
+	// See: https://github.com/kubernetes/kubernetes/pull/52570/files/3eea91793dfbc3fdb0799589fac3790c4cde58a4#r140391019
+	// Version of the manifest.
+	// ManifestVersion string `json:"manifestVersion,omitempty" yaml:"manifestVersion,omitempty"`
+
+	// NamePrefix will prefix the names of all resources mentioned in the manifest
+	// including generated configmaps and secrets.
+	NamePrefix string `json:"namePrefix,omitempty" yaml:"namePrefix,omitempty"`
+
+	// Labels to add to all objects and selectors.
+	// These labels would also be used to form the selector for apply --prune
+	// Named differently than “labels” to avoid confusion with metadata for this object
+	ObjectLabels map[string]string `json:"objectLabels,omitempty" yaml:"objectLabels,omitempty"`
+
+	// Annotations to add to all objects.
+	ObjectAnnotations map[string]string `json:"objectAnnotations,omitempty" yaml:"objectAnnotations,omitempty"`
+
+	// A Base entry specifies the relative paths within the package.
+	// It could be any format that kubectl -f allows, i.e. files, directories, URLs and globs.
+	Bases []string `json:"bases,omitempty" yaml:"bases,omitempty"`
+
+	// An Overlay entry is very similar to an Base entry.
+	// It specifies the relative paths within the package, and could be any format that kubectl -f allows.
+	// It should be able to be merged by Strategic Merge Patch on top of its corresponding Base if any.
+	Overlays []string `json:"overlays,omitempty" yaml:"overlays,omitempty"`
+
+	// List of configmaps to generate from configuration sources.
+	// Base/overlay concept doesn't apply to this field.
+	// If a configmap want to have a base and an overlay, it should go to Bases and Overlays fields.
+	Configmaps []ConfigMap `json:"configmaps,omitempty" yaml:"configmaps,omitempty"`
+
+	// List of secrets to generate from secret sources.
+	// Base/overlay concept doesn't apply to this field.
+	// If a secret want to have a base and an overlay, it should go to Bases and Overlays fields.
+	Secrets []Secret `json:"secrets,omitempty" yaml:"secrets,omitempty"`
+
+	// Whether PersistentVolumeClaims should be deleted with the other resources.
```

Comment:
This seems like something that should be declared per-pvc, not "globally".  Why isn't this done via an overlay to add/remove the usual annotations/options to specific PVCs?

Reply:
This is a convenience mechanism. Per PVC would be less convenient and, as you point out, could be accomplished other ways.